### ISSUE TITLE
Better if there is a way to identify the constraint annotated types

### ISSUE THREAD

**Issue by chamil321** [ballerina-platform/ballerina-library#3112]
**Description:**
$subject otherwise, all data binded values go through the `constraint:validate(payload);` function which is unnecessary execution.

```ballerina
if (constraint:required(typedesc<Person>) {
    Person validPayload = check constraint:validate(payload);
}
```

**Comment by ldclakmal** [comment]
@chamil321 I agree this API make sense from constraint package users point of view. But, as per the runtime APIs available, we are checking the annotations availability and their values recursively in the constraint package. We process that against the constrained value, only if the constraint annotations are available. There is no API to read the annotations of all the types which are being referred at a single point. Therefore, we are checking it level by level in a recursive manner.

So, in order to provide the suggested API in constraint package, we have to recursively check the annotations of all the referred types. That will take almost same time as the `constraint:validate` API take. With that, if we have the above sample use case, for the values that the constraint is required, will take double amount of time.
